From cece3fb64927c943debdd0d09f264ba3a5546be5 Mon Sep 17 00:00:00 2001
Message-Id: <cece3fb64927c943debdd0d09f264ba3a5546be5.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 2 Aug 2014 09:39:33 +0530
Subject: [PATCH 241/312] MUST_REBASE [VPG]: Use new connection_mutex while
 calling set/get backlight

!FOR_UPSTREAM: DPST feature is not to be upstreamed, as the corresponding
User mode component can not be open sourced.

commit 51fd371bbaf94018a1223b4e2cf20b9880fd92d4
Author: Rob Clark <robdclark@gmail.com>
Date:   Tue Nov 19 12:10:12 2013 -0500

    drm: convert crtc and connection_mutex to ww_mutex (v5)

Change-Id: I6d56b57ade8eeb07580dbec55defe0a2455e5496
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_dpst.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dpst.c b/drivers/gpu/drm/i915/i915_dpst.c
index 1698845..433ee20 100644
--- a/drivers/gpu/drm/i915/i915_dpst.c
+++ b/drivers/gpu/drm/i915/i915_dpst.c
@@ -170,11 +170,11 @@ i915_dpst_set_user_enable(struct drm_device *dev, bool enable)
 		dev_priv->dpst.saved.is_valid = false;
 
 		/* Avoid warning messages */
-		mutex_lock(&dev->mode_config.mutex);
+		drm_modeset_lock(&dev->mode_config.connection_mutex, NULL);
 		if (dev_priv->dpst.enabled)
 			ret = i915_dpst_disable_hist_interrupt(dev);
 
-		mutex_unlock(&dev->mode_config.mutex);
+		drm_modeset_unlock(&dev->mode_config.connection_mutex);
 	}
 
 	return ret;
@@ -274,11 +274,11 @@ i915_dpst_restore_luma(struct drm_device *dev)
 	dev_priv->dpst.blc_adjustment = dev_priv->dpst.saved.blc_adjustment;
 
 	/* Avoid warning messages */
-	mutex_lock(&dev->mode_config.mutex);
+	drm_modeset_lock(&dev->mode_config.connection_mutex, NULL);
 	mutex_lock(&dev_priv->backlight_lock);
 	i915_dpst_set_brightness(dev, panel->backlight.level);
 	mutex_unlock(&dev_priv->backlight_lock);
-	mutex_unlock(&dev->mode_config.mutex);
+	drm_modeset_unlock(&dev->mode_config.connection_mutex);
 
 	/* IE mod table entries are saved in the hardware even if the table
 	 * is disabled, so we only need to re-enable the table */
-- 
1.7.9.5

